internal/config/config.go

```diff
@@ -380,6 +390,34 @@ func (t *Tunnel) SetDefaults() {
 	}
 }
 
+func hasMultiRegistrySupport(framework, version string) bool {
+	minVersions := map[string]string{
+		"cypress":               "12.14.0",
+		"playwright-cucumberjs": "1.35.1",
+		"playwright":            "1.35.1",
+		"testcafe":              "2.6.2",
+	}
+
+	v, ok := minVersions[framework]
+	if !ok {
+		return true
+	}
+	maxVersion := semver.MustParse(v)
+	curVersion, err := semver.NewVersion(version)
+	if err != nil {
+		return true
+	}
+	return curVersion.GreaterThan(maxVersion)
+}
+
+// SetDefaults updates npm default values
+func (n *Npm) SetDefaults(framework, version string) {
+	if n.Registry != "" && hasMultiRegistrySupport(framework, version) {
+		log.Warn().Msg("npm.registry has been deprecated, please use npm.registries instead")
```

Comment:
Framework version could be `package.json` here, I think, since we haven't resolved the specific version at this point in the execution. `hasMultiRegistrySupport` will return `true` in this case and you *might* show this warning incorrectly. I don't think we have to resolve the version here, but maybe reserve the warning for when we're absolutely sure the recommended action won't break the user?

Reply:
That's probably true. I was at first thinking future proof-ness, but as you mentioned it may bring some kind of wrong warning. Currently there is ~3.5% of user using that value (but without knowing the actual content).
I agree we should keep the behaviour assuming that the `package.json` contains an older version.

The new question will be: when should we reassess that behaviour ?
Probably in a couple of month when we have above 50% usage on "supported" versions.
I'll create a ticket as a reminder on this.